### PR TITLE
Add docs about mocking non existent functions

### LINKED ISSUE #464: Consider adding a note about the possibility of applying a cheatcode to a contract/function that doesn't exist
Refers to `roll`, `prank`, `warp` cheatcodes

### PR DESCRIPTION
**Pull request by cptartur**
Closes #464 

## Introduced changes

- Added a section to `start_mock_call` about mocking functions that are non-existent
- Updated the info about mocks not working for function calls within a contract

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: docs/src/appendix/cheatcodes/start_mock_call.md [hunk 1]
```diff
@@ -2,17 +2,28 @@
...
+> Mocks do not have any effect on function calls withing the contract itself.
+> E.g. for a function withing a contract defined like this:
+>
+> ```rust
+> fn function_a(self: @ContractState) {
+>   function_b()
+> } 
+> ```
+>
+> Mocking `function_b` would have no effect.
```
Comment (piotmag769) [inline docs/src/appendix/cheatcodes/start_mock_call.md:26]:
```suggestion
> mocking `function_b` would have no effect when calling `function_a`.
```

File: docs/src/appendix/cheatcodes/start_mock_call.md [hunk 2]
```diff
@@ -44,12 +55,73 @@ We can use `start_mock_call` in a test to change the data returned by `get_balan
...
 }
-```
\ No newline at end of file
+```
+
+## Mocking non-existent functions
+
+It is also possible to simulate contract having a specific method by mocking a non-existent selector.
+
+Let's assume we have defined a interface to use a dispatcher, but this interface contains a function that is not
```
Comment (piotmag769) [inline docs/src/appendix/cheatcodes/start_mock_call.md:72]:
```suggestion
Let's assume we have defined an interface to use a dispatcher, but this interface contains a function that is not
actually defined in actually deployed contract:
```

File: docs/src/appendix/cheatcodes/start_mock_call.md [hunk 3]
```diff
@@ -44,12 +55,73 @@ We can use `start_mock_call` in a test to change the data returned by `get_balan
...
+}
+
+#[external(v0)]
+impl IContractImpl of IContract<ContractState> {
+    fn call_not_actually_implemented(self: @ContractState) -> felt252 {
+        // ...
+        let other_contract_dispatcher = IOtherContractDispatcher { contract_address };
+        let result = other_contract_dispatcher.function_not_actually_implemented();
+        
+        result
```
Comment (piotmag769) [inline docs/src/appendix/cheatcodes/start_mock_call.md:95]:
```suggestion
        other_contract_dispatcher.function_not_actually_implemented()
```

File: docs/src/appendix/cheatcodes/start_mock_call.md [hunk 4]
```diff
@@ -44,12 +55,73 @@ We can use `start_mock_call` in a test to change the data returned by `get_balan
...
-```
\ No newline at end of file
+```
+
+## Mocking non-existent functions
+
+It is also possible to simulate contract having a specific method by mocking a non-existent selector.
+
+Let's assume we have defined a interface to use a dispatcher, but this interface contains a function that is not
+actually defined in actually deployed contract:
```
Comment (war-in) [inline docs/src/appendix/cheatcodes/start_mock_call.md:73]:
```suggestion
actually defined in a deployed contract:
```

File: docs/src/appendix/cheatcodes/start_mock_call.md [hunk 5]
```diff
@@ -44,12 +55,73 @@ We can use `start_mock_call` in a test to change the data returned by `get_balan
...
+
+```rust
+// ...
+// Assume we define an interface like that, but it is incorrect.
```
Comment (war-in) [inline docs/src/appendix/cheatcodes/start_mock_call.md:77]:
```suggestion
// Assume we define an interface like that, but it is extended.
```
Why would anyone define an incorrect interface?

File: docs/src/appendix/cheatcodes/start_mock_call.md (hunk 1 above)
Comment (cptartur) [reply docs/src/appendix/cheatcodes/start_mock_call.md:26]: It's below a snipped and we have `:` at the end of the previous sentence so I think it should start with capital letter?

File: docs/src/appendix/cheatcodes/start_mock_call.md (hunk 5 above)
Comment (cptartur) [reply docs/src/appendix/cheatcodes/start_mock_call.md:77]: I guess not, I meant that it contains a function that was not actually implemented in the main contract. I'll reword it so it explains it better.

**Review by war-in — APPROVED** [review]

**Pull request merged**